File: FunnyAlgorithms-master/Java/8 - Utilities/Binary search.java
Dado un arreglo ordenado ascendentemente de tamaño n, busca el elemento x y devuelve su posición, si no lo encuentra devuelve -1.

static int binary_search(int array[], int x){
	int l = 0, r = arr.length-1;
	while (l <= r) {
		int m = (l+r)/2;
		if(array[m] < x) l = m+1;
		else if (array[m] > x) r = m-1;
		else return m;
	}
	return -1;
}

File: algorithms/Binary Search/binary_search.rb
def binary_search(n, arr)
    middle = arr.length / 2
    i = 0
    j = arr.length - 1
  
    while i < j
      if arr[middle] == n
        return true
      elsif arr[middle] < n
        i = middle + 1
        middle = i + j / 2
      else
        j = middle - 1
        middle = i + j / 2
      end
    end
    false
  end